aten/src/ATen/native/cuda/KernelUtils.cuh

```diff
@@ -62,17 +133,24 @@ __device__ __forceinline__ void fastSpecializedAtomicAdd(
     __half2 value2;
     value2.x = static_cast<__half>(value);
     value2.y = __int2half_rz(0);
-    atomicAdd(reinterpret_cast<__half2*>(target_addr), value2);
+    ATOMICADD(reinterpret_cast<__half2*>(target_addr), value2);
 
   } else if (!low_byte && index > 0) {
     __half2 value2;
     value2.x = __int2half_rz(0);
     value2.y = static_cast<__half>(value);
-    atomicAdd(reinterpret_cast<__half2*>(target_addr - 1), value2);
+    ATOMICADD(reinterpret_cast<__half2*>(target_addr - 1), value2);
 
   } else {
-    atomicAdd(
-        reinterpret_cast<__half*>(tensor) + index, static_cast<__half>(value));
+#ifdef USE_ROCM
+    gpuAtomicAddNoReturn(
+        reinterpret_cast<at::Half*>(tensor) + index,
+        static_cast<at::Half>(value));
+#else
+    atomicAddNoReturn(
```

Comment:
`atomicAdd(` ?

Internally this reports the compilation error:

```
1 error detected in the compilation of "fbcode/caffe2/aten/src/ATen/native/cuda/AveragePool3d.cu".
fbcode/caffe2/aten/src/ATen/native/cuda/KernelUtils.cuh(150): error: identifier "atomicAddNoReturn" is undefined
          detected during:
            instantiation of "void at::native::fastSpecializedAtomicAdd(scalar_t *, index_t, index_t, scalar_t) [with scalar_t=c10::Half, index_t=int, <unnamed>=(void *)nullptr]" 
(226): here
            instantiation of "void at::native::fastAtomicAdd(scalar_t *, index_t, index_t, scalar_t, __nv_bool) [with scalar_t=c10::Half, index_t=int]" 
fbcode/caffe2/aten/src/ATen/native/cuda/AveragePool3d.cu(267): here
            instantiation of "void at::native::<unnamed>::avg_pool3d_cuda_update_grad_input_atomic<scalar_t,accscalar_t>(at::PackedTensorAccessor64<const scalar_t, 4UL, at::DefaultPtrTraits>, at::PackedTensorAccessor64<scalar_t, 4UL, at::DefaultPtrTraits>, int, int, int, int, int, int, int, int, int, __nv_bool, int, int, int) [with scalar_t=c10::Half, accscalar_t=float]" 
fbcode/caffe2/aten/src/ATen/native/cuda/AveragePool3d.cu(560): here
```

Reply:
Still need to resolve this issue: this `atomicAddNoReturn` caused the failure in the internal phabrictor test suites.

Reply:
Fixed now.